soc/arm/st_stm32/common/soc_config.c

```diff
@@ -60,13 +60,38 @@ static int st_stm32_common_config(void)
 	LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_DBGMCU);
 #endif /* LL_APB1_GRP1_PERIPH_DBGMCU */
 
+#endif /* CONFIG_USE_SEGGER_RTT */
+
+
+#if defined(CONFIG_STM32_ENABLE_DEBUG_SLEEP_STOP)
+
 #if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32MP1X)
 	HAL_EnableDBGSleepMode();
-#else
+#else /* CONFIG_SOC_SERIES_STM32H7X || CONFIG_SOC_SERIES_STM32MP1X */
+
+#if defined(SOC_SERIES_STM32G0X)
+	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_DBGMCU);
+	LL_DBGMCU_EnableDBGStopMode();
+	LL_APB1_GRP1_DisableClock(LL_APB1_GRP1_PERIPH_DBGMCU);
+#elif defined(SOC_SERIES_STM32L0X)
+	LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_DBGMCU);
+	LL_DBGMCU_EnableDBGStopMode();
+	LL_APB2_GRP1_DisableClock(LL_APB2_GRP1_PERIPH_DBGMCU);
+#else /* all other parts */
+	/* Enable the Debug Module during all and any Low power mode */
 	LL_DBGMCU_EnableDBGStopMode();
+#endif
+
 #endif /* CONFIG_SOC_SERIES_STM32H7X || CONFIG_SOC_SERIES_STM32MP1X */
 
-#endif /* CONFIG_USE_SEGGER_RTT */
+#else /* CONFIG_STM32_ENABLE_DEBUG_SLEEP_STOP */
+
+#if defined(SOC_SERIES_STM32G4X)
+	/* keep in mind that debugging draws a lot of power */
+	LL_DBGMCU_DisableDBGStopMode();
+#endif /*defined(SOC_SERIES_STM32G4X) */
```

Comment:
Let's use this PR to fix this inconsistency between series.
I think this behavior can be extended to all series. If we don't agree on that, let's remove on G4 as well.

Reply:
I think we can agree, I've checked like this:

rlezuo@rlezuo-desktop:~/projects/zephyr/modules/hal/stm32/stm32cube$ find . -name "*ll_system.h" -exec grep -H  "void LL_DBGMCU_DisableDBGStopMode" '{}' \;
./stm32c0xx/drivers/include/stm32c0xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32f0xx/drivers/include/stm32f0xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32f1xx/drivers/include/stm32f1xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32f2xx/drivers/include/stm32f2xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32f3xx/drivers/include/stm32f3xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32f4xx/drivers/include/stm32f4xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32f7xx/drivers/include/stm32f7xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32g0xx/drivers/include/stm32g0xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32g4xx/drivers/include/stm32g4xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32h5xx/drivers/include/stm32h5xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32l0xx/drivers/include/stm32l0xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32l1xx/drivers/include/stm32l1xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32l4xx/drivers/include/stm32l4xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32l5xx/drivers/include/stm32l5xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32u5xx/drivers/include/stm32u5xx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32wbaxx/drivers/include/stm32wbaxx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32wbxx/drivers/include/stm32wbxx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
./stm32wlxx/drivers/include/stm32wlxx_ll_system.h:__STATIC_INLINE void LL_DBGMCU_DisableDBGStopMode(void)
rlezuo@rlezuo-desktop:~/projects/zephyr/modules/hal/stm32/stm32cube$ find . -name "*ll_system.h" -exec grep -L  "void LL_DBGMCU_DisableDBGStopMode" '{}' \;
./common_ll/include/stm32_ll_system.h
./stm32h7xx/drivers/include/stm32h7xx_ll_system.h
./stm32mp1xx/drivers/include/stm32mp1xx_ll_system.h

Reply:
ok thanks. I guess this should be symmetric with `Enable()` functions